sol-did/programs/sol-did/src/integrations/is_authority.rs

```diff
@@ -0,0 +1,384 @@
+}
+
+pub fn derive_did_account(authority: &Pubkey) -> (Pubkey, u8) {
+    Pubkey::find_program_address(
+        &[DID_ACCOUNT_SEED.as_bytes(), authority.key().as_ref()],
+        &id(),
+    )
+}
+
+pub fn derive_did_account_with_bump(authority: &Pubkey, bump_seed: u8) -> Result<Pubkey> {
+    Pubkey::create_program_address(
+        &[
+            DID_ACCOUNT_SEED.as_bytes(),
+            authority.key().as_ref(),
+            &[bump_seed],
+        ],
+        &id(),
+    )
+    .map_err(|_| Error::from(ErrorCode::ConstraintSeeds))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::constants::VM_DEFAULT_FRAGMENT_NAME;
+    use crate::state::{DidAccount, VerificationMethodFlags};
+    use crate::VerificationMethod;
+    use std::cell::RefCell;
+    use std::rc::Rc;
+    use std::str::FromStr;
+
+    fn create_test_authority() -> Pubkey {
+        Pubkey::new_unique()
+    }
+
+    fn create_test_did(test_authority: Pubkey) -> DidAccount {
+        DidAccount {
+            version: 0,
+            bump: 0,
+            nonce: 0,
+            initial_verification_method: VerificationMethod {
+                fragment: VM_DEFAULT_FRAGMENT_NAME.to_string(),
+                flags: VerificationMethodFlags::CAPABILITY_INVOCATION.bits(),
+                method_type: 0,
+                key_data: test_authority.to_bytes().to_vec(),
+            },
+            verification_methods: vec![],
+            services: vec![],
+            native_controllers: vec![],
+            other_controllers: vec![],
+        }
+    }
+
+    #[test]
+    fn test_derive_did_account() {
+        let authority = Pubkey::from_str("6TE7bGggnzahkE7Snfyi8M4LuB3D4YV8CjoBJxn8UDsY").unwrap();
+        let expected_did_account =
+            Pubkey::from_str("3spWJgYRKqrZnBkgv6dwjohKG5x3ZBEdxoLxuC2LfwD2").unwrap();
+        let expected_bump = 255;
+
+        let (did_account_pubkey, bump) = derive_did_account(&authority);
+
+        assert_eq!(did_account_pubkey, expected_did_account);
+        assert_eq!(bump, expected_bump);
+    }
+
+    #[test]
+    fn test_derive_did_account_with_bump() {
+        let authority = Pubkey::from_str("6TE7bGggnzahkE7Snfyi8M4LuB3D4YV8CjoBJxn8UDsY").unwrap();
+        let expected_did_account =
+            Pubkey::from_str("3spWJgYRKqrZnBkgv6dwjohKG5x3ZBEdxoLxuC2LfwD2").unwrap();
+        let expected_bump = 255;
+
+        let did_account_pubkey = derive_did_account_with_bump(&authority, expected_bump).unwrap();
+
+        assert_eq!(did_account_pubkey, expected_did_account);
+    }
+
+    #[test]
+    fn test_is_authority() {
+        let test_authority = create_test_authority();
+        let test_did_account = create_test_did(test_authority);
+        let mut data: Vec<u8> = Vec::with_capacity(1024);
+        test_did_account.try_serialize(&mut data).unwrap();
+        let derived_did_account = derive_did_account(&test_authority);
+
+        let mut lamports = 1; // must be > 0 to pass the Account::try_from
```

Comment:
@dankelleher I'm removing all these comments, because the do not seem to hold true (anymore?)